In [1]:
report_card = {"Student_names":["Vanshika","Vivaan","Siddhant","Mitansh","Kareena"],
               "physics_grades":[38,45,65,75,80],
               "chemistry_grades":[100,45,56,67,43],
               "maths_grades":[90,80,70,60,50],
               "biology_grades":[10,20,30,40,50]
              }
report_card

{'Student_names': ['Vanshika', 'Vivaan', 'Siddhant', 'Mitansh', 'Kareena'],
 'physics_grades': [38, 45, 65, 75, 80],
 'chemistry_grades': [100, 45, 56, 67, 43],
 'maths_grades': [90, 80, 70, 60, 50],
 'biology_grades': [10, 20, 30, 40, 50]}

In [2]:
#setup
import pandas as pd

In [3]:
#making the report card dictionary into dataframe
report_card_df = pd.DataFrame(report_card)
report_card_df

Student_names  physics_grades  chemistry_grades  maths_grades  \
0      Vanshika              38               100            90   
1        Vivaan              45                45            80   
2      Siddhant              65                56            70   
3       Mitansh              75                67            60   
4       Kareena              80                43            50   

   biology_grades  
0              10  
1              20  
2              30  
3              40  
4              50

In [4]:
#First 3 values from the dataframe
report_card_df.head(3)

Student_names  physics_grades  chemistry_grades  maths_grades  \
0      Vanshika              38               100            90   
1        Vivaan              45                45            80   
2      Siddhant              65                56            70   

   biology_grades  
0              10  
1              20  
2              30

In [5]:
#3 random values from the dataframe
report_card_df.sample(3)

Student_names  physics_grades  chemistry_grades  maths_grades  \
4       Kareena              80                43            50   
1        Vivaan              45                45            80   
2      Siddhant              65                56            70   

   biology_grades  
4              50  
1              20  
2              30

In [6]:
#Describing all the numeric values in the dataframe
report_card_df.describe()

physics_grades  chemistry_grades  maths_grades  biology_grades
count        5.000000          5.000000      5.000000        5.000000
mean        60.600000         62.200000     70.000000       30.000000
std         18.420098         23.209912     15.811388       15.811388
min         38.000000         43.000000     50.000000       10.000000
25%         45.000000         45.000000     60.000000       20.000000
50%         65.000000         56.000000     70.000000       30.000000
75%         75.000000         67.000000     80.000000       40.000000
max         80.000000        100.000000     90.000000       50.000000

# Insights
 - what is the average chemistry grade? 62.2
 - what is the average biology grade? 30,0
 - what is the is the lowest math grade? 50.0
 - what is the highest physics grade? 80.0
 - How many scores are there in the math coloumn? 5

In [7]:
# Making a total grade coloumn
report_card_df["total_grade"] = report_card_df["physics_grades"]+ report_card_df["chemistry_grades"]+ report_card_df["maths_grades"]+ report_card_df["biology_grades"]
report_card_df

Student_names  physics_grades  chemistry_grades  maths_grades  \
0      Vanshika              38               100            90   
1        Vivaan              45                45            80   
2      Siddhant              65                56            70   
3       Mitansh              75                67            60   
4       Kareena              80                43            50   

   biology_grades  total_grade  
0              10          238  
1              20          190  
2              30          221  
3              40          242  
4              50          223

In [8]:
report_card_df["average_grade"] = report_card_df["total_grade"]/4
report_card_df

Student_names  physics_grades  chemistry_grades  maths_grades  \
0      Vanshika              38               100            90   
1        Vivaan              45                45            80   
2      Siddhant              65                56            70   
3       Mitansh              75                67            60   
4       Kareena              80                43            50   

   biology_grades  total_grade  average_grade  
0              10          238          59.50  
1              20          190          47.50  
2              30          221          55.25  
3              40          242          60.50  
4              50          223          55.75

In [9]:
report_card_df.describe()

physics_grades  chemistry_grades  maths_grades  biology_grades  \
count        5.000000          5.000000      5.000000        5.000000   
mean        60.600000         62.200000     70.000000       30.000000   
std         18.420098         23.209912     15.811388       15.811388   
min         38.000000         43.000000     50.000000       10.000000   
25%         45.000000         45.000000     60.000000       20.000000   
50%         65.000000         56.000000     70.000000       30.000000   
75%         75.000000         67.000000     80.000000       40.000000   
max         80.000000        100.000000     90.000000       50.000000   

       total_grade  average_grade  
count     5.000000       5.000000  
mean    222.800000      55.700000  
std      20.486581       5.121645  
min     190.000000      47.500000  
25%     221.000000      55.250000  
50%     223.000000      55.750000  
75%     238.000000      59.500000  
max     242.000000      60.500000

In [10]:
report_card_insights_df = report_card_df.describe().transpose()
report_card_insights_df

count   mean        std    min     25%     50%    75%    max
physics_grades      5.0   60.6  18.420098   38.0   45.00   65.00   75.0   80.0
chemistry_grades    5.0   62.2  23.209912   43.0   45.00   56.00   67.0  100.0
maths_grades        5.0   70.0  15.811388   50.0   60.00   70.00   80.0   90.0
biology_grades      5.0   30.0  15.811388   10.0   20.00   30.00   40.0   50.0
total_grade         5.0  222.8  20.486581  190.0  221.00  223.00  238.0  242.0
average_grade       5.0   55.7   5.121645   47.5   55.25   55.75   59.5   60.5

In [11]:
report_card_insights_df.to_csv("Report card.csv")

In [12]:
report_card_df.loc[report_card_df["total_grade"]>=140, "Status"]="Pass"
report_card_df.loc[report_card_df["total_grade"]<140, "Status"]="Fail"
report_card_df

Student_names  physics_grades  chemistry_grades  maths_grades  \
0      Vanshika              38               100            90   
1        Vivaan              45                45            80   
2      Siddhant              65                56            70   
3       Mitansh              75                67            60   
4       Kareena              80                43            50   

   biology_grades  total_grade  average_grade Status  
0              10          238          59.50   Pass  
1              20          190          47.50   Pass  
2              30          221          55.25   Pass  
3              40          242          60.50   Pass  
4              50          223          55.75   Pass

In [13]:
report_card_insight_df = report_card_df
report_card_insight_df.to_csv("Report card.csv")

In [14]:
report_card_df.loc[(report_card_df["average_grade"] >=90), "Character_grade"]="A"
report_card_df.loc[(report_card_df["average_grade"] >=70) & (report_card_df["average_grade"]<90),"Character_grade"]="B"
report_card_df.loc[(report_card_df["average_grade"] >=50) & (report_card_df["average_grade"]<69),"Character_grade"]="C"
report_card_df.loc[(report_card_df["average_grade"] >=40) & (report_card_df["average_grade"]<49),"Character_grade"]="D"
report_card_df.loc[(report_card_df["average_grade"] <40), "Character_grade"]= "F"

In [15]:
report_card_df

Student_names  physics_grades  chemistry_grades  maths_grades  \
0      Vanshika              38               100            90   
1        Vivaan              45                45            80   
2      Siddhant              65                56            70   
3       Mitansh              75                67            60   
4       Kareena              80                43            50   

   biology_grades  total_grade  average_grade Status Character_grade  
0              10          238          59.50   Pass               C  
1              20          190          47.50   Pass               D  
2              30          221          55.25   Pass               C  
3              40          242          60.50   Pass               C  
4              50          223          55.75   Pass               C

In [16]:
report_card_df.to_csv("Report Card.csv")